pkg/gofr/datasource/sql/sql.go

```diff
@@ -159,18 +160,24 @@ func getDBConfig(configs config.Config) *DBConfig {
 		MaxIdleConn: maxIdleConn,
 		// only for postgres
 		SSLMode: configs.GetOrDefault("DB_SSL_MODE", "disable"),
+		Charset: configs.Get("DB_CHARSET"),
 	}
 }
 
 func getDBConnectionString(dbConfig *DBConfig) (string, error) {
 	switch dbConfig.Dialect {
 	case "mysql":
-		return fmt.Sprintf("%s:%s@tcp(%s:%s)/%s?charset=utf8&parseTime=True&loc=Local&interpolateParams=true",
+		if dbConfig.Charset == "" {
+			dbConfig.Charset = "utf8"
+		}
```

Comment:
While I like your change, I think you somehow spotted an existing issue.

It might be considered as a breaking changes, but I would expect the default value to be `utf8mb4`

But it's more a question for @aryanmehrotra @vipul-rawat @srijan-27 @Umang01-hash that could and should be addressed in another PR.

Except that, your PR is fine and could be merged as-is.

Reply:
@ccoVeille I agree with your suggestion to change the default value to `utf8mb4`as it supports a broader range of Unicode characters and aligns with modern applications needs.